e2e-tests/src/test/button_game/mod.rs

```diff
@@ -5,18 +5,115 @@ use anyhow::Result;
 use assert2::{assert, let_assert};
 use helpers::sign;
 use log::info;
+use sp_core::Pair;
 
 use crate::{
     test::button_game::helpers::{
-        assert_recv, assert_recv_id, refute_recv_id, setup_button_test, wait_for_death,
-        ButtonTestContext,
+        assert_recv, assert_recv_id, mega, refute_recv_id, setup_button_test, setup_dex_test,
+        wait_for_death, ButtonTestContext, DexTestContext,
     },
     Config,
 };
 
 mod contracts;
 mod helpers;
 
+/// Test trading on simple_dex.
+///
+/// The scenario does the following (given 3 tokens A, B, C):
+///
+/// 1. Enables A <-> B, and A -> C swaps.
+/// 2. Adds (A, 2000M), (B, 5000M), (C, 10000M) of liquidity.
+/// 3. Makes a swap A -> B and then B -> A for the amount of B received in the first swap.
+/// 4. Checks that the price after the two swaps is the same as before (with a dust allowance of 1 for rounding).
+/// 5. Checks that it's possible to make an A -> C swap, but impossible to make a C -> A swap.
+pub fn simple_dex(config: &Config) -> Result<()> {
+    let DexTestContext {
+        conn,
+        authority,
+        account,
+        dex,
+        token1,
+        token2,
+        token3,
+        mut events,
+    } = setup_dex_test(config)?;
+    let authority_conn = &sign(&conn, &authority);
+    let account_conn = &sign(&conn, &account);
+    let token1 = token1.as_ref();
+    let token2 = token2.as_ref();
+    let token3 = token3.as_ref();
+    let dex = dex.as_ref();
+
+    dex.add_swap_pair(authority_conn, token1.into(), token2.into())?;
```

Comment:
pairs with wAzero shoudl be covered too, I reckon this is what will see most traffic

Reply:
As I said in another comment - I don't think it matters at all for these tests, because the tokens are treated generically. I guess you're more worried about testing wAzero itself? I would say that should be a separate test scenario, where we mostly test wrapping and unwrapping.

Reply:
we can do it like that - it would boil down to testing wrap/unwrap so not too involved I suppose

Reply:
I'll do it in another PR